### PR TITLE
Fix unbounded null throwing exception

### LINKED ISSUE #40: Bug: null numeric unbounded arguments try thrown IllegalArgumentException
When unbounded arguments have a null value, the command line parser try to thrown an `CommandLineException.OutOfRangeArgumentValue` but because there are no boundaries it throws an `IllegalArgumentException`.

### PR DESCRIPTION
**Pull request by magicDGS**
- Fixes #40 
- Add previously failing test

### CODE REVIEW

**Comment by magicDGS** [comment]
Could you please have a look to this @cmnbroad?  It is a very simple fix for a bug that is preventing me to update my dependencies to the latest GATK/Barclay versions... Thank you in advance.

**Comment by cmnbroad** [comment]
Hi @magicDGS Yes, will take a look tomorrow.

**Comment by magicDGS** [comment]
Thank you very much @cmnbroad!

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java [hunk 1]
```diff
@@ -1186,6 +1187,11 @@ public boolean isFlag(){
             return field.getType().equals(boolean.class) || field.getType().equals(Boolean.class);
         }
 
+        /** Returns {@code true} if the argument has a bounded range; {@code false} otherwise. */
+        public boolean hasBoundedRange() {
+            return this.minValue != Double.NEGATIVE_INFINITY || this.maxValue != Double.POSITIVE_INFINITY;
+        }
+
```
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: I think this could be private for now. Also, now that this method exists, we should use it in the other places where we do the same check (i.e., in the argdef constructor). Likewise, we should add a hasRecommendedRange and use that at the appropriate call sites.

File: src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java [hunk 2]
```diff
@@ -841,6 +841,9 @@ public void testWithDoubleBoundariesArgumentsForInteger() {
         // recommended values are not explicitly verified by the tests, but do force the code through the warning code paths
         @Argument(doc = "Integer in the range [0, 30]", optional = true, minValue = 0, minRecommendedValue = 10, maxRecommendedValue = 15, maxValue = 30)
         public int integerArg = 20;
+        // Integer without boundaries and null should be allowed
+        @Argument(doc = "Integer with null value allowed", optional = true)
+        public Integer nullInteger = null;
     }
```
Comment (cmnbroad) [inline src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java]: Its confusing to have this argument, and the corresponding test, live inside the "WithBoundariesArguments" class and test. It should live somewhere else, maybe in its own test if there is no current test thats appropriate. We should also add a similar test case for a Double (with null) arg.

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java [hunk 3]
```diff
@@ -535,7 +535,7 @@ private void checkArgumentRange(final ArgumentDefinition argumentDefinition, fin
         final Double argumentDoubleValue = (argumentValue == null) ? null : ((Number)argumentValue).doubleValue();
 
         // Check hard limits first, if specified
-        if (isOutOfRange(argumentDefinition.minValue, argumentDefinition.maxValue, argumentDoubleValue)) {
+        if (argumentDefinition.hasBoundedRange() && isOutOfRange(argumentDefinition.minValue, argumentDefinition.maxValue, argumentDoubleValue)) {
```
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: We should add a hasRecommendedRange method, similar to hasBoundedRange, and use that below where we check the recommended range.

Comment (magicDGS) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: It is not necessary for the logging path, because if there is no range there are no logging. Anyway, I included the check and added an exception (should never reach).

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java (hunk 1 above)
Comment (magicDGS) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: Done.

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java [hunk 4]
```diff
@@ -552,6 +552,7 @@ private void checkArgumentRange(final ArgumentDefinition argumentDefinition, fin
                 logger.warn("Argument --{} has value {}, but maximum recommended is {}",
                         argumentDefinition.getLongName(), argumentDoubleValue, argumentDefinition.maxRecommendedValue);
             }
+            // if it does not have any recommended value, it does not log anything as expected
```
Comment (magicDGS) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: I removed the comment and added an exception, because now it should not happen.

Comment (magicDGS) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: Sorry, this should be the comment. I misunderstood where it is....

Comment (magicDGS) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java]: Done.

**Comment by magicDGS** [comment]
I addressed your comments in the last commit. Thank you very much @cmnbroad -- back to you

**Comment by magicDGS** [comment]
I think that this is ready to go, @cmnbroad. It is important for me to update dependencies with the new changes in some of my projects. Thank you in advance!

**Comment by magicDGS** [comment]
Thanks for this!
